cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1604,19 +1623,145 @@ String zkConnectionString(int connectToReplicas, Function<Integer, String> zkNod
+         */
+        protected Future<Config> brokerConfig(Admin ac, int brokerId) {
+            ConfigResource resource = new ConfigResource(ConfigResource.Type.BROKER, String.valueOf(brokerId));
+            return kafkaFutureToVertxFuture(reconciliation, ac.describeConfigs(singletonList(resource)).values().get(resource));
+        }
+
+        protected Future<Admin> adminClient(AdminClientProvider adminClientProvider, String namespace, String cluster, int podId) {
+            Future<Secret> clusterCaCertSecretFuture = secretOperations.getAsync(
+                    namespace, KafkaResources.clusterCaCertificateSecretName(cluster));
+            Future<Secret> coKeySecretFuture = secretOperations.getAsync(
+                    namespace, ClusterOperator.secretName(cluster));
+            String hostname = KafkaCluster.podDnsName(namespace, cluster, KafkaCluster.kafkaPodName(cluster, podId)) + ":" + KafkaCluster.REPLICATION_PORT;
+
+            return CompositeFuture.join(clusterCaCertSecretFuture, coKeySecretFuture).compose(compositeFuture -> {
+                Secret clusterCaCertSecret = compositeFuture.resultAt(0);
+                if (clusterCaCertSecret == null) {
+                    return Future.failedFuture(Util.missingSecretException(namespace, KafkaCluster.clusterCaCertSecretName(cluster)));
+                }
+                Secret coKeySecret = compositeFuture.resultAt(1);
+                if (coKeySecret == null) {
+                    return Future.failedFuture(Util.missingSecretException(namespace, ClusterOperator.secretName(cluster)));
+                }
+
+                Promise<Admin> blockingPromise = Promise.promise();
+                vertx.createSharedWorkerExecutor("kubernetes-ops-pool").executeBlocking(
+                    future -> {
+                        try {
+                            future.complete(adminClientProvider.createAdminClient(hostname, clusterCaCertSecret, coKeySecret, "cluster-operator"));
+                        } catch (RuntimeException e) {
+                            future.fail(e);
+                        }
+                    }, blockingPromise);
+                return blockingPromise.future();
+            });
+        }
+
+        /**
+         * If the kafka pod is ready, change its configuration dynamically if both necessary and possible.
+         * If it is not possible set the flag for specific podID an it will be rolled in {@code kafkaRollingUpdate}.
+         * @return Succeeded future
+         */
+        Future<ReconciliationState> kafkaBrokerDynamicConfiguration() {
+            // patches scale up/down situations
+            int runningBeforeAndAfterReplicas = Math.min(oldKafkaReplicas, kafkaCluster.getReplicas());
+            List<Future> configFutures = new ArrayList<>(runningBeforeAndAfterReplicas);
+
+            for (int podId = 0; podId < runningBeforeAndAfterReplicas; podId++) {
```

Comment:
This method will try to update all the brokers concurrently. That's might be fast when everything goes well, but if the change does something which breaks the brokers we're basically bringing down the cluster. Off the top of my head I don't know which configs can be changed dynamically (obviously it's a subset of the Kafka dynamic configs) to decide how possible/likely it is for a user to break their cluster in this way. However, the algorithm probably needs to do something similar to the `KafkaRoller`, updating each broker sequentially and performing some kind of check after an update that the broker is still OK.